app/views/posts/_form.html.erb

```diff
@@ -17,15 +17,15 @@
         <div class="col-sm-12">
           <div class="create-post m-3">
             <div class="form-group">
-              <%= form.text_field :title, class: "form-control "+is_error(@post,(:title)).to_s, placeholder: :title, maxlength: 20%>
+              <%= form.text_field :title, class: style_error_class(@post, :title, "form-control"), placeholder: :title, maxlength: 20%>
               <p class="text-danger"><%= error_msg(@post, :title)  %></p>
             </div>
           </div>
           <div class="create-post m-3">
             <div class="form-group">
-              <%= form.rich_text_area :body, class: "textarea "+is_error(@post,(:body)).to_s, placeholder: "Text(optional)" %>
+              <%= form.rich_text_area :body, class: style_error_class(@post, :body, "textarea"), placeholder: "Text(optional)" %>
               <p class="text-danger"><%= error_msg(@post, :body)  %></p>
-              <%= form.file_field :images, multiple: true, :class=>"form-control "+is_error(@post,(:images)).to_s, :placeholder=>"" %>
+              <%= form.file_field :images, multiple: true, class: style_error_class(@post, :images, "form-control"), :placeholder=>"" %>
```

Comment:
thhis style_error_class seems to be called in 10 different places which means that loop you have inside of the helper will be run 10 times which is a perf problem

Reply:
Refactored code in helper method